src/applications/validate-mhv-account/containers/ErrorMessage.jsx

```diff
@@ -9,33 +9,47 @@ import NeedsVAPatient from '../components/errors/NeedsVAPatient';
 import CreateMHVAccountFailed from './CreateMHVAccountFailed';
 import UpgradeAccountFailed from './UpgradeAccountFailed';
 import { ACCOUNT_STATES } from './../constants';
+import { MVI_ERROR_STATES } from './../../../platform/monitoring/RequiresMVI/constants';
 
 export default class ErrorMessage extends React.Component {
   render() {
     const { params } = this.props;
 
-    // Replace hyphens with underscores to match ACCOUNT_STATES constants
-    const errorCode = params.errorCode
+    let errorCode = params.errorCode
       ? params.errorCode.replace(/-/g, '_')
-      : '';
+      : undefined;
 
-    // Render error messaging based on code
-    switch (errorCode) {
-      case ACCOUNT_STATES.DEACTIVATED_MHV_IDS:
-        return <DeactivatedMHVId />;
-      case ACCOUNT_STATES.MULTIPLE_IDS:
-        return <MultipleMHVIds />;
-      case ACCOUNT_STATES.NEEDS_SSN_RESOLUTION:
-        return <NeedsSSNResolution />;
-      case ACCOUNT_STATES.NEEDS_VA_PATIENT:
-        return <NeedsVAPatient />;
-      case ACCOUNT_STATES.REGISTER_FAILED:
-        return <CreateMHVAccountFailed />;
-      case ACCOUNT_STATES.UPGRADE_FAILED:
-        return <UpgradeAccountFailed />;
-      default:
-        return <GenericError />;
+    if (errorCode) {
+      if (errorCode.startsWith('mvi_error')) {
```

Comment:
it might be a little more concise to just combine these 2 checks and then put everything in a single switch statement, i.e.
```javascript
if (errorCode && errorCode.startsWith('mvi_error')) {
   errorCode = errorCode.replace('mvi_error_', '').toUpperCase();
}

// no else statement

switch (errorCode) {
   case MVI_ERROR_STATES.NOT_FOUND:
       return <NeedsSSNResolution />;
   /* all the other cases */
    default: 
        return <GenericError />
```